diff --git a/3rdParty/rocksdb/rocksdb/CMakeLists.txt b/3rdParty/rocksdb/rocksdb/CMakeLists.txt
index 5b5e08b..5d0bfc3 100644
--- a/3rdParty/rocksdb/rocksdb/CMakeLists.txt
+++ b/3rdParty/rocksdb/rocksdb/CMakeLists.txt
@@ -174,7 +174,7 @@ if (CMAKE_COMPILER_IS_GNUCC)
   else()
     set(ROCKSDB_MALLOC_USABLE_SIZE "-DROCKSDB_MALLOC_USABLE_SIZE")
   endif()
-  set(BASE_FLAGS "${BASE_FLAGS} -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer -fno-builtin-memcmp -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers  ${ROCKSDB_MALLOC_USABLE_SIZE} -isystem -fPIC -D${OS_DEFINE} ${POSIX_FLAG}")
+  set(BASE_FLAGS "${BASE_FLAGS} -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -fno-omit-frame-pointer -fno-builtin-memcmp -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers  ${ROCKSDB_MALLOC_USABLE_SIZE} -isystem -fPIC -D${OS_DEFINE} ${POSIX_FLAG}")
 
   set(CMAKE_C_FLAGS                "-g"                             CACHE INTERNAL "default C compiler flags")
   set(CMAKE_C_FLAGS_DEBUG          "-O0 -g -Werror"                         CACHE INTERNAL "C debug flags")
diff --git a/lib/Basics/ArangoGlobalContext.h b/lib/Basics/ArangoGlobalContext.h
index 64eccce..75d3852 100644
--- a/lib/Basics/ArangoGlobalContext.h
+++ b/lib/Basics/ArangoGlobalContext.h
@@ -25,6 +25,10 @@
 
 #include "Basics/Common.h"
 
+#ifdef __arm__
+  #include "Basics/FileUtils.h"
+#endif
+
 namespace arangodb {
 class ArangoGlobalContext {
  public:
