diff --git a/3rdParty/rocksdb/rocksdb/CMakeLists.txt b/3rdParty/rocksdb/rocksdb/CMakeLists.txt
index 54fca1d..ff012ad 100644
--- a/3rdParty/rocksdb/rocksdb/CMakeLists.txt
+++ b/3rdParty/rocksdb/rocksdb/CMakeLists.txt
@@ -168,13 +168,30 @@ if (CMAKE_COMPILER_IS_GNUCC)
   if (VERBOSE)
     message(STATUS "Compiler type GNU: ${CMAKE_CXX_COMPILER}")
   endif ()
+
+# "-momit-leaf-frame-pointer" not on ARM
+  SET(OPT_momit_leaf_frame_pointer "-momit-leaf-frame-pointer")
+
+  if(CMAKE_TARGET_ARCHITECTURES MATCHES "armv7"
+  OR CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "armv7l" )
+    SET(OPT_momit_leaf_frame_pointer "")
+  endif()
+
+  if (VERBOSE)
+    message(STATUS "CMAKE_TARGET_ARCHITECTURES:" ${CMAKE_TARGET_ARCHITECTURES})
+    message(STATUS "CMAKE_HOST_SYSTEM_PROCESSOR:" ${CMAKE_HOST_SYSTEM_PROCESSOR})
+    message(STATUS "OPT_momit_leaf_frame_pointer:" ${OPT_momit_leaf_frame_pointer})
+  endif ()
+
+
 # -DSNAPPY -DZLIB
   if (SOLARIS OR DARWIN)
     set(ROCKSDB_MALLOC_USABLE_SIZE "")
   else()
     set(ROCKSDB_MALLOC_USABLE_SIZE "-DROCKSDB_MALLOC_USABLE_SIZE")
   endif()
-  set(BASE_FLAGS "${BASE_FLAGS} -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer -fno-builtin-memcmp -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers  ${ROCKSDB_MALLOC_USABLE_SIZE} -isystem -fPIC -D${OS_DEFINE} ${POSIX_FLAG}")
+  set(BASE_FLAGS "${BASE_FLAGS} -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -fno-omit-frame-pointer ${OPT_momit_leaf_frame_pointer} -fno-builtin-memcmp -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers  ${ROCKSDB_MALLOC_USABLE_SIZE} -isystem -fPIC -D${OS_DEFINE} ${POSIX_FLAG}")
+
 
   set(CMAKE_C_FLAGS                "-g"                             CACHE INTERNAL "default C compiler flags")
   set(CMAKE_C_FLAGS_DEBUG          "-O0 -g -Werror"                         CACHE INTERNAL "C debug flags")
diff --git a/lib/Basics/HybridLogicalClock.cpp b/lib/Basics/HybridLogicalClock.cpp
index bf055a5..4cb082a 100644
--- a/lib/Basics/HybridLogicalClock.cpp
+++ b/lib/Basics/HybridLogicalClock.cpp
@@ -26,7 +26,7 @@
 char arangodb::basics::HybridLogicalClock::encodeTable[65] 
     = "-_ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
 
-char arangodb::basics::HybridLogicalClock::decodeTable[256]
+signed char arangodb::basics::HybridLogicalClock::decodeTable[256]
     = {-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,        //   0 - 15
        -1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,        //  16 - 31
        -1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1, 0,-1,-1,        //  32 - 47
diff --git a/lib/Basics/HybridLogicalClock.h b/lib/Basics/HybridLogicalClock.h
index 4151ac2..022d8ad 100644
--- a/lib/Basics/HybridLogicalClock.h
+++ b/lib/Basics/HybridLogicalClock.h
@@ -127,7 +127,7 @@ class HybridLogicalClock {
     }
     uint64_t r = 0;
     for (size_t i = 0; i < len; i++) {
-      char c = decodeTable[static_cast<uint8_t>(p[i])];
+      signed char c = decodeTable[static_cast<uint8_t>(p[i])];
       if (c < 0) {
         return 0;
       }
@@ -159,7 +159,7 @@ class HybridLogicalClock {
 
   static char encodeTable[65];
 
-  static char decodeTable[256];
+  static signed char decodeTable[256];
 
 };
 
